### PR TITLE
Refactor bundle

### PR DESCRIPTION
**Pull request by dbu**
huge refactoring that builds on #15 with a radical rewrite of everything.

TODO
* [ ] Tests
* [ ] Update documentation for latest refactoring
* [ ] Check if CSE works too

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,187 +1,170 @@
...
+Provided for you are:
 
+* A controller to render a search box and the search page with twig templates
+* A service to query google site search
 * A service which provides paging for the search results
 
-* A controller service which can be used to render a search box from a template
+### Built-in Search Engines Support
 
+For now, only [Google site search](http://www.google.com/sitesearch/) is
```
Comment (ralf57) [inline README.md]: +1

**Comment by dbu** [comment]
@lsmith77 would you have time to have a look at this until thursday? it ends up between a refactoring and a rewrite...

File: README.md [hunk 2]
```diff
@@ -1,187 +1,170 @@
...
 
-
 Create a route for the search action. The easiest is to just use the provided routing.yml from your main project routing.yml
 
-``` yaml
-liip_search:
-    resource: "@LiipSearchBundle/Resources/config/routing.yml"
-```
+    liip_search:
+        resource: "@LiipSearchBundle/Resources/config/routing.yml"
```
Comment (lsmith77) [inline README.md]: should we then also move this to XML

Comment (lsmith77) [inline README.md]: I guess this applies to all of the service config files too

**Comment by lsmith77** [comment]
its a bit tricky for me to review this .. especially as there were never tests in this Bundle. can you explain the goal of the refactoring/rewrite?

**Comment by dbu** [comment]
i started looking into this and basically, the interface did not really abstract the search, so that in the end there was a controller for google search specifically. that was not purpose of this bundle.

with the refactoring, i try to simplify the whole thing, with the idea of a very simple array format for search results that can be rendered with the twig templates.

i can update the README tonight to give usage examples and the like to make it more clear.

File: README.md (hunk 2 above)
Comment (dbu) [inline README.md]: yep, agreed

**Comment by dbu** [comment]
i just updated the readme. the thing really is just the SearchInterface. but i did some more digging and essentially we probably should write a [pagerfanta](https://github.com/whiteoctober/pagerfanta) adapter for the google custom search api - or while are at throwing away things, make that a wrapper for [google-apis-php-clients](https://github.com/StephenMelrose/google-apis-php-clients) instead of our own implementation.

@thePanz did you look at those google-apis-php-clients by @stephenmelrose? the description says that the clients are not ready, but there are no open issues on the repository and its been there for a while. @stephenmelrose how stable is your google custom search api client at this point? any chance for a release?

**Comment by dbu** [comment]
closed in favor of #17

**Comment by stephenmelrose** [comment]
> @stephenmelrose how stable is your google custom search api client at this point? any chance for a release?

To be honest, I don't really have the time or the will power to maintain this properly as an open source project. If you'd like to take over it, I'd be happy to transfer ownership.

All I would say is I think Google now provide and maintain their own PHP API Client, although it doesn't contain custom search _yet_. See https://developers.google.com/api-client-library/php/start/installation.

**Comment by dbu** [comment]
thanks for the feedback @stephenmelrose ! i think we will just go with doing that call ourselves. the call is not that complicated - the harder part is transforming the results to the format we want to use, and there is no library that can help us there.

**Comment by stephenmelrose** [comment]
Feel free to just lift that part out of mine if want.

**Pull request closed without merging**